Code/Source/Entity/ActorEntityManager.cpp

```diff
@@ -0,0 +1,157 @@
+#include <EMotionFX/Source/TransformData.h>
+#include <Entity/ActorEntityManager.h>
+#include <Utilities/RGLUtils.h>
+#include <rgl/api/core.h>
+
+namespace RGL
+{
+    ActorEntityManager::ActorEntityManager(AZ::EntityId entityId)
+        : EntityManager(entityId)
+    {
+        EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusConnect(entityId);
+    }
+
+    ActorEntityManager::ActorEntityManager(ActorEntityManager&& other)
+        : EntityManager(std::move(other))
+    {
+        EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusDisconnect();
+    }
+
+    ActorEntityManager::~ActorEntityManager()
+    {
+        for (MeshPair mesh : m_meshes)
+        {
+            RGL_CHECK(rgl_mesh_destroy(mesh.m_rglMesh));
+        }
+    }
+
+    void ActorEntityManager::Update()
+    {
+        UpdateMeshVertices();
+        EntityManager::Update();
+    }
+
+    void ActorEntityManager::OnActorInstanceCreated(EMotionFX::ActorInstance* actorInstance)
+    {
+        m_actorInstance = actorInstance;
+        EMotionFX::Actor* actor = actorInstance->GetActor();
+
+        static constexpr size_t LodLevel = 0UL; // Highest LOD.
+        const size_t NodeCount = actor->GetNumNodes();
+        for (size_t jointIndex = 0LU; jointIndex < NodeCount; ++jointIndex)
+        {
+            EMotionFX::Mesh* mesh = actor->GetMesh(LodLevel, jointIndex);
+            rgl_mesh_t rglMesh;
+            if (mesh && (rglMesh = EMotionFXMeshToRglMesh(*mesh)))
+            {
+                m_meshes.push_back({ mesh, rglMesh });
+            }
+        }
+
+        m_entities.reserve(m_meshes.size());
+        for (MeshPair& mesh : m_meshes)
+        {
+            rgl_entity_t entity = nullptr;
+            Utils::SafeRglEntityCreate(entity, mesh.m_rglMesh);
+            if (entity)
+            {
+                m_entities.emplace_back(entity);
+            }
+        }
+
+        if (!m_entities.empty())
+        {
+            UpdatePose();
+        }
+    }
+
+    void ActorEntityManager::UpdateMeshVertices()
+    {
+        m_actorInstance->UpdateMeshDeformers(0.0f);
+
+        for (MeshPair& mesh : m_meshes)
+        {
+            UpdateVertexPositions(*mesh.m_eMotionMesh);
+            RGL_CHECK(rgl_mesh_update_vertices(mesh.m_rglMesh, m_positions.data(), m_positions.size()));
+        }
+    }
+
+    void ActorEntityManager::UpdateVertexPositions(const EMotionFX::Mesh& mesh)
+    {
+        const size_t VertexCount = mesh.GetNumVertices();
+        auto* vertices = static_cast<const AZ::Vector3*>(mesh.FindVertexData(EMotionFX::Mesh::ATTRIB_POSITIONS));
+
+        m_positions.clear();
+        m_positions.reserve(VertexCount);
+        for (size_t vertex = 0; vertex < VertexCount; ++vertex)
+        {
+            m_positions.push_back(Utils::RglVector3FromAzVec3f(vertices[vertex]));
+        }
+    }
+
+    AZStd::vector<rgl_vec3i> ActorEntityManager::CollectIndexData(const EMotionFX::Mesh& mesh)
```

Comment:
I'd rename it to `GetIndexData` or `GetMeshIndices` to be consistent with the previous comment.